packages/block-library/src/button/editor.scss

```diff
@@ -10,14 +10,17 @@
 }
 
 .wp-block-button {
-	display: inline-block;
-	margin-bottom: 0;
 	position: relative;
 
 	[contenteditable] {
 		cursor: text;
 	}
 
+	// Don't let placeholder expand parent width.
```

Comment:
When this was applied to the button itself, it prevented the margin collapse. By moving it down here, the net result is the same: an empty button isn't 100% wide, but it looks the same in frontend and backend.